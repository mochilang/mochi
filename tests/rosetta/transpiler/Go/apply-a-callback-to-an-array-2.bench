{
  "duration_us": 488,
  "memory_bytes": 1168,
  "name": "main"
}