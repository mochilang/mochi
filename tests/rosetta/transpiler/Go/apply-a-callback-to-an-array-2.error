exit status 1
# command-line-arguments
../../../tests/rosetta/transpiler/Go/apply-a-callback-to-an-array-2.go:29:10: print(fmt.Sprint((i * i))) (no value) used as value