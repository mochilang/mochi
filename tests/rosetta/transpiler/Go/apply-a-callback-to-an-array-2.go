//go:build ignore

// Generated by Mochi v0.10.38 on 2025-07-23 23:44:40 GMT+7
package main

import (
	"fmt"
)

func each(xs []int, f func(int) any) {
	for _, x := range xs {
		f(x)
	}
}

func Map(xs []int, f func(int) int) []int {
	var r []int = []int{}
	_ = r
	for _, x := range xs {
		r = append(r, f(x))
	}
	return r
}

func mochiMain() {
	var s []int = []int{1, 2, 3, 4, 5}
	_ = s
	each(s, func(i int) any {
		return print(fmt.Sprint((i * i)))
	})
	fmt.Println(fmt.Sprint(Map(s, func(i int) int {
		return (i * i)
	})))
}

func main() {
	mochiMain()
}
