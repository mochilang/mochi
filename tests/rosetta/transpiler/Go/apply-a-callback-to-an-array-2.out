1491625[1 4 9 16 25]